src/components/CheckBox/CheckBoxGroup.tsx

```diff
@@ -106,14 +107,16 @@ export const CheckBoxGroup: FC<CheckboxGroupProps> = React.forwardRef(
             checked={value?.includes(item.value)}
             key={item.value}
             onChange={() => {
-              const optionIndex = value?.indexOf(item.value);
-              const newValue = [...value];
-              if (optionIndex === -1) {
-                newValue.push(item.value);
-              } else {
-                newValue.splice(optionIndex, 1);
+              if (value) {
+                const optionIndex: number = value?.indexOf(item.value);
```

Comment:
whats with the optional chaining? Seems like if thats necessary this could possibly set optionIndex to undefined and screwup your if/else statement. I'm guessing its not actually necessary?